core/domain/config_validators.py

```diff
@@ -74,9 +74,11 @@ def _get_external_id_relationships(cls, item):
                 'config_property_ids',
                 config_models.ConfigPropertyModel,
                 [item.id[:item.id.rfind(base_models.VERSION_DELIMITER)]]),
-            base_model_validators.ExternalModelFetcherDetails(
-                'committer_ids',
-                user_models.UserSettingsModel, [item.committer_id])]
+            base_model_validators.UserSettingsModelFetcherDetails(
+                'committer_ids', [item.committer_id],
+                may_contain_system_ids=True,
```

Comment:
I don't believe either of these cases is possible, because only admins set these. Shouldn't it be False for both?

Reply:
`may_contain_system_ids` should probably be `False` but when user that is/was previously admin requests deletion and they modified some config in history it can be pseudo ID then

Reply:
@seanlip  ConfigPropertySnapshotMetadataModel with id starting with oppia_csrf_secret is being created but the committer id for this is set to system user at https://github.com/oppia/oppia/blob/75b0ef8148e0aca3ef22278dc762aa5349f2e1da/core/controllers/base.py#L547
So this means that ConfigPropertySnapshotMetadataModel has models with committers id as "admin". So I think may_contain_system_ids should also be True.

Reply:
OK, these rationales seem fine, let's keep it as you have it. Thanks for verifying!